CommonMark/Parser/BlockMethods.cs

```diff
@@ -92,16 +93,355 @@ private static void BreakOutOfLists(ref Block blockRef, LineInfo line)
+
+                        if (row.LastChild == null)
+                        {
+                            row.FirstChild = row.LastChild = cell;
+                        }
+                        else
+                        {
+                            row.LastChild.NextSibling = cell;
+                            row.LastChild = cell;
+                        }
+
+                        cell.IsOpen = false;
+                    }
+                }
+            }
+        }
+
+        static void MakeTableRows(Block table, StringBuilder sb)
+        {
+            var asStr = table.StringContent.ToString();
+            var lines = asStr.Split('\n');
+
+            var offset = 0;
+
+            for (var i = 0; i < lines.Length; i++)
+            {
+                var line = lines[i];
+
+                var lineLength = line.Length;
+                var hasLineBreak = offset + lineLength < asStr.Length && asStr[offset + lineLength] == '\n';
+                if (hasLineBreak) lineLength++;
+
+                // skip the header row
+                if (i != 1 && !string.IsNullOrWhiteSpace(line))
+                {
+                    var rowStartsInDocument = table.SourcePosition + offset;
+                    var row = new Block(BlockTag.TableRow, rowStartsInDocument);
+                    row.SourceLastPosition = rowStartsInDocument + lineLength;
+
+                    row.StringContent = new StringContent();
+                    row.StringContent.Append(asStr, offset, row.SourceLength);
+
+                    if (table.LastChild == null)
+                    {
+                        table.FirstChild = row;
+                        table.LastChild = row;
+                    }
+                    else
+                    {
+                        table.LastChild.NextSibling = row;
+                        table.LastChild = row;
+                    }
+
+                    MakeTableCells(row, sb);
+                    row.IsOpen = false;
+                }
+
+                offset += lineLength;
+            }
+        }
+
+        static bool TryMakeTable(Block b, LineInfo line, CommonMarkSettings settings)
+        {
+            if ((settings.AdditionalFeatures & CommonMarkAdditionalFeatures.GithubStyleTables) == 0) return false;
+
+            var asStr = b.StringContent.ToString();
+            var lines = asStr.Split('\n');
+
+            if (lines.Length < 2) return false;
+
+            var sb = new StringBuilder();
+
+            var columnsLine = ParseTableLine(lines[0], sb);
+            if (columnsLine.Count == 1) return false;
+
+            var headerLine = ParseTableLine(lines[1], sb);
+            if (headerLine.Count == 1) return false;
+
+            var headerAlignment = new List<TableHeaderAlignment>();
+
+            foreach (var headerPart in headerLine)
+            {
+                var trimmed = headerPart.Trim();
+                if (trimmed.Length < 3) return false;
```

Comment:
empty header cells are removed later in this method but here they will cause the method to return.